File: LeetCode 26 Remove Element from Duplicate Array/RemoveDuplicateElementFromArray.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        
        res = 1  # 1️⃣ Result/Writer pointer. Index 1 se start karenge kyunki index 0 fix hai.
        
        # 2️⃣ Loop start hota hai index 1 se (compare karne ke liye)
        for i in range(1, len(nums)):
            
            # 3️⃣ Logic: Kya current number (i) apne pichle number (i-1) se alag hai?
            if nums[i] != nums[i - 1]:
                nums[res] = nums[i] # 4️⃣ Agar alag hai, toh usse 'res' position pe likh do
                res += 1 # 5️⃣ Writer pointer ko aage badhao
                
        return res # 6️⃣ 'res' already count maintain kar raha hai

File: LeetCode 26 Remove Element from Duplicate Array/RemoveDuplicateElementFromArray.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int i = 0; // 1️⃣ Slow Pointer (Writer) - Unique elements ka track rakhta hai

        // 2️⃣ Fast Pointer (Reader) - Array ko scan karta hai start se end tak
        for (int j = 1; j < nums.length; j++) {
            
            // 3️⃣ Logic: Agar Reader (j) ko jo number mila, wo Writer (i) ke number se alag hai?
            if (nums[j] != nums[i]) {
                i++; // 4️⃣ Writer ko agli khali seat par bhejo
                nums[i] = nums[j]; // 5️⃣ Wahan par naya unique number copy karo
            }
        }
        
        // 6️⃣ Kyunki index 0 se start hota hai, length 'i + 1' hogi
        return i + 1;
    }
}